tests/external/notifications/aws/test_aws_notification_handler.py

```diff
@@ -0,0 +1,208 @@
+    """Test Cases for AWSNotificationHandler."""
+
+    @mock_sns
+    def test_confirm_subscription(self):
+        # Setup
+        conn = boto3.client('sns', region_name='us-east-1')
+        response = conn.create_topic(Name='CostUsageNotificationDemo')
+        headers = [('X-Amz-Sns-Message-Type', 'SubscriptionConfirmation'),
+                   ('X-Amz-Sns-Message-Id', '010b7c42-3f24-4865-83b1-13120d79466b'),
+                   ('x-Amz-Sns-Subscription-Arn', 'arn:aws:sns:us-west-2:123456789012:MyTopic:2bcfbf39-05c3-41de-beaa-fcfcc21c8f55'),
+                   ('Content-Length', 1637),
+                   ('Content-Type', 'text/plain; charset=UTF-8'),
+                   ('Host', 'masu.net:6868'),
+                   ('Connection', 'Keep-Alive'),
+                   ('User-Agent', 'Amazon Simple Notification Service Agent'),
+                   ('Accept-Encoding', 'gzip,deflate')]
+        headers.append(('X-Amz-Sns-Topic-Arn', response['TopicArn']))
+
+        body_dict = {"Type" : "SubscriptionConfirmation",
+                     "MessageId" : "010b7c42-3f24-4865-83b1-13120d79466b",
+                     "Token" : "2336412f37fb687f5d51e6e241da92fd72d769c811cb257f2ca8bbb9e329dd2eb427b0935ce7fe84f91e14e5994d2d9039b3aade5b4f16dd5ab3979b05751962ceed2c7e193e7b7797630c31eec29477949064522feb104a311b44e1268ba7bdfdc06d7e6f39174dbb69326a379d06cbbc96adcdc24939e36892e024aa0e96d2",
+                     "TopicArn" : response['TopicArn'],
+                     "Message" : "You have chosen to subscribe to the topic arn:aws:sns:us-east-1:589179999999:CostUsageNotificationDemo.\nTo confirm the subscriptionDataBuilder, visit the SubscribeURL included in this message.",
+                     "SubscribeURL" : "https://sns.us-east-1.amazonaws.com/?Action=ConfirmSubscription&TopicArn=arn:aws:sns:us-east-1:589179999999:CostUsageNotificationDemo&Token=2336412f37fb687f5d51e6e241da92fd72d769c811cb257f2ca8bbb9e329dd2eb427b0935ce7fe84f91e14e5994d2d9039b3aade5b4f16dd5ab3979b05751962ceed2c7e193e7b7797630c31eec29477949064522feb104a311b44e1268ba7bdfdc06d7e6f39174dbb69326a379d06cbbc96adcdc24939e36892e024aa0e96d2",
+                     "Timestamp" : "2018-07-05T12:29:27.641Z",
+                     "SignatureVersion" : "1",
+                     "Signature" : "L8VmRMT55sv326Rt1HBawA8hPKR0plaBfvI8hlZNm2nhaMkVzvFkvKiMZucfQ7bYWaYqVF+xzvqkaRnMw8/uC4QcrM5DK5HcXYfAJCeI8t32jTg3Kcdn3SGBiyxcBWYt8qSXL7403ho4vIVpCHRuX6K2ZydS7sxx6umr/nanaOUGnkDeC91GFoUMslNw7yITc0YunPwg8GcvKgLhsQr2d7C5I3zDprA+Ds29voLN6v66hH0uR0LEfDq9Ky0Dgii4UebYUgKBCWSy5ETsRHqug5tPNzv9afexjdXrjEUYbvYZO+RRDPztwGjyHw6CL35B85FLZ7UDPLeRysZq47gb+Q==",
+                     "SigningCertURL" : "https://sns.us-east-1.amazonaws.com/SimpleNotificationService-eaea6120e66ea12e88dcd8bcbddca752.pem"}
+        body = json.dumps(body_dict)
+        try:
+            _ = AWSNotificationHandler(headers, body, validation=False)
+        except Exception:
+            self.fail('Unexpected exception')
```

Comment:
Why is this try/except block here? Won't the test fail if an exception is thrown anyway?